### PR TITLE
Updates for MPB 0.4

### PR DESCRIPTION
**Pull request by JackDunnNZ**
Ref #22 and https://github.com/JuliaOpt/MathProgBase.jl/pull/91
Can't merge until CoinOptServices updates to support MPB 0.4 otherwise tests break.

@mlubin It's pretty simple, but the macro to generate the wrapper functions seems a bit hacky. Any ideas for a better way?

It passes tests locally after fixing the conflicts with CoinOptServices re. supporting MPB 0.4.

### CODE REVIEW

File: src/AmplNLWriter.jl [hunk 1]
```diff
@@ -135,24 +135,70 @@ type AmplNLMathProgModel <: AbstractMathProgModel
...
+end
+
+# Macro to generate wrappers for SolverInterface functions
+# e.g.
+#   @wrap status(m::AmplNLMathProgModel) = m.status
+# produces
+#   status(m::AmplNLMathProgModel) = m.status
+#   status(outer::AmplNLNonlinearModel) = status(outer.inner)
+#   status(outer::AmplNLLinearQuadraticModel) = status(outer.inner)
+macro wrap(args)
```
Comment (mlubin) [inline src/AmplNLWriter.jl]: I would do something like:
```
for f in [:status,:getvartype,:getsense, ...]
    @eval $f(m::AmplNLNonlinearModel) = $f(m.inner)
    @eval $f(m::AmplNLLinearQuadraticModel) = $f(m.inner)
end
```
It's a bit more manual but less fancy.

**Comment by mlubin** [comment]
CC @tkelman

File: src/AmplNLWriter.jl (hunk 1 above)
Comment (JackDunnNZ) [inline src/AmplNLWriter.jl]: It's a bit of a pain for the ones with extra parameters though, unless there's a smart way?

Comment (mlubin) [inline src/AmplNLWriter.jl]: Won't another loop for the 2-argument wrappers cover everything?

Comment (JackDunnNZ) [inline src/AmplNLWriter.jl]: Does that keep type information?

Comment (mlubin) [inline src/AmplNLWriter.jl]: It's not a big deal if that's missing, you'll still get an error message

Comment (JackDunnNZ) [inline src/AmplNLWriter.jl]: Okay cool, that'll be much better then

**Comment by JackDunnNZ** [comment]
Much better now, thanks!

**Comment by tkelman** [comment]
I don't really have time right now to deal with fixing CoinOptServices, so the fastest option would be taking the BinDeps script and making new Bonmin.jl and Couenne.jl stubs that just install the solvers at first. Bonmin has a C API now (might not be in a stable release yet, need to check), and I'm planning on eventually doing something specific for Couenne, either via Cxx or special purpose C wrappers depending on when I get around to it.

**Comment by mlubin** [comment]
@JackDunnNZ, travis is passing now that CoinOptServices is bumped. Could you restart the appveyor build?

**Comment by JackDunnNZ** [comment]
Thanks for tagging it, I restarted appveyor just now

**Comment by mlubin** [comment]
All green now :+1:

**Pull request merged**